doc/developers/maintainer.rst.template

```diff
@@ -217,7 +190,7 @@ Reference Steps
 
         In addition if on merging, the last commit, containing the `[cd build]` marker,
         is empty, the CD jobs won't be triggered. In this case, you can directly push
-        a commit with the marker in the `{{ version_short }}.X` to trigger them.
+        a commit with the marker in the `{{ version_short }}.X` branch to trigger them.
 
     - If the steps above went fine, proceed **with caution** to create a new tag for the
```

Comment:
OK so something that I did not realised in the past: when creating the release via the GitHub interface, when clicking on "Publish Release", a notification is sent to people following the release. Usually, a message is sent while the documentation, the wheels, and the conda-forge builds might not be ready.

I think this is worth adding a note or a warning regarding this step:

- we push the tag via the CLI and only publish the release via the GitHub interface once we build the wheels, the conda binaries, and deploy the documentation
- alternatively, we can create the tag via the GitHub interface but we should only click on "Save Draft" without publishing yet the release.

What I'm not sure is if the "Save Draft" option is creating the tag directly or if it is pending. If this is the latter then this is not a viable solution because we need the tag to be able to create the conda-forge builds.

Reply:
> Usually, a message is sent while the documentation, the wheels, and the conda-forge builds might not be ready.

The publish release item happens after uploading the wheels so wheels are ready at that time. For my part, I usually publish the release after merging the conda forge PR but it's true that there's a small delay for conda-forge because after merging there's a intermediate staging step before the package is available on the public channel. Regarding the doc, since we start the conda-forge builds after merging the release PR, the doc should be ready when we merge the conda forge PR, but we can say that we should wait a bit before publishing and make sure the updated web site is live.